File: LeetCode/1423/main.cpp
class Solution {
public:
  int maxScore(vector<int> &arr, int k) {
    int n = arr.size();
    int sum = 0, minSum = 0;
    int left = 0, total = 0;

    // Need to find a subarray of lenth (n - k) with minimum sum ?

    for (int i = 0; i < n - k; ++i) {
      sum += arr[i];
      total += arr[i];
    }

    minSum = sum;

    for (int right = n - k; right < n; ++right) {

      sum += arr[right];
      sum -= arr[left];
      left++;

      minSum = min(minSum, sum);
      total += arr[right];
    }

    return total - minSum;
  }
};

File: leetcode/1423/main.go
package main

import "fmt"

func maxScore(arr []int, k int) int {
	n := len(arr)

	left, s, totalSum, minSum := 0, 0, 0, 0

	for i := 0; i < n-k; i++ {
		s += arr[i]
		totalSum += arr[i]
	}

	minSum = s

	for right := n - k; right < n; right++ {
		s += arr[right]
		s -= arr[left]
		left++

		minSum = min(minSum, s)
		totalSum += arr[right]
	}

	return totalSum - minSum

}
